mailserver/mailserver.go

```diff
@@ -225,9 +241,16 @@ func (s *WMailServer) exceedsPeerRequests(peer []byte) bool {
 
 // processRequest processes the current request and re-sends all stored messages
 // accomplishing lower and upper limits.
-func (s *WMailServer) processRequest(peer *whisper.Peer, lower, upper uint32, bloom []byte) []*whisper.Envelope {
-	ret := make([]*whisper.Envelope, 0)
-	var err error
+func (s *WMailServer) processRequest(peer *whisper.Peer, lower, upper uint32, bloom []byte) (ret []*whisper.Envelope, err error) {
+	// Recover from possible goleveldb panics
+	defer func() {
```

Comment:
@PombeirP ended up moving this logic to a [shared function](https://github.com/status-im/status-go/pull/1061/files#diff-f422bb8b8162ca6a8f7fbff615267548R159) would be nice if we could use the same

Reply:
@adriacidre yes I tried to use the same but in this case we need to return an error. It can be generalised maybe but it would be less readable I think